File: Competitive Programming/codeforces/training/D_Plus_Minus_Permutation.py
# Copyright (c) 2023, Le Duc Phuc Long

# If you don't think twice, you have to code twice.

# Import session
import sys
input = sys.stdin.readline
from collections import defaultdict

############ ---- Input Functions ---- ############
def inp():
    return int(input())

def inlt():
    return list(map(int, input().split()))

def instr():
    return list(input().strip())

def invr():
    return map(int, input().split())

############ ---- Other Functions ---- ############
def gcd(a, b):
    while b:
        a, b = b, a % b
    return a

def compute_sequence(l, r):
    n = r - l + 1
    a = l
    d = 1

    sequence_sum = (n * (2 * a + (n - 1) * d)) // 2

    return sequence_sum

# Precompute
    
# Main function
for _ in range(inp()):
    n, x, y = invr()
    lcm = x*y//gcd(x, y)
    plus = n // x - n//lcm
    minus = n // y - n//lcm
    print(compute_sequence(n-plus+1, n)-compute_sequence(1, minus))
    

File: Competitive Programming/codeforces/training/D_Plus_Minus_Permutation.cpp
// Copyright (c) 2023, Le Duc Phuc Long

/*If you don't think twice, you have to code twice.*/

// Include
#include <bits/stdc++.h>  // NOLINT
// #include <conio.h>

// Namespace
using namespace std;  // NOLINT

// Define
#define nln '\n'

// Typedef and Struct
typedef int64_t lo;

// Constant

// Declare global variables.
vector<pair<lo, lo>> dir4 = {{0, 1}, {0, -1}, {1, 0}, {-1, 0}};
// End of global variable declaration.

void precompute() {
}

lo sum_sequence(lo l, lo r, lo s) {
    return (l+r)*((r-l)/s+1)/2ll;
}

lo __lcm(lo x, lo y) {
    return x*y/__gcd(x, y);
}

void solve() {
    lo n, x, y; cin >> n >> x >> y;
    lo q_c = n/__lcm(x, y), q_x = n/x - q_c, q_y = n/y - q_c; 
    cout << sum_sequence(n-q_x+1, n, 1) - sum_sequence(1, q_y, 1) << nln;
    // cout << "OK" << nln;
}

int main(int argc, char* argv[]) {
    //freopen("in.txt", "r", stdin);
    //freopen("out.txt", "w", stdout);
    cin.tie(0)->sync_with_stdio(0);
    cout.tie(0)->sync_with_stdio(0);
    int T = 1;
    cin >> T;
    precompute();
    while (T--) {
        solve();
    }
    cerr << "It's ok Long, Good for now !!" << nln;
    return 0;
}